notebooks/intercomparison-covermaps.ipynb

```diff
@@ -0,0 +1,1346 @@
+{
```

Comment:
<p>Okay so I see you create a new dataframe for each predicted map, merge the test points to the predicted map and calculate the metrics on that dataframe.</p><p>This approach makes sense!</p><p>One small drawback is that it makes comparison between the other maps difficult. For example do Glad and ESA have the same mistakes?</p><p>Instead of creating multiple dataframes I suggest adding to one single dataframe and then computing all metrics on that one single dataframe:</p><p> </p><pre class="ql-syntax" spellcheck="false">esa = ee.ImageCollection("ESA/WorldCover/v100")
esa_results = esa.filterBounds(test_coll).map(lambda x: rasterExtraction(x, 10, test_coll)).flatten()
esa_sampled = geemap.ee_to_gdf(esa_results)
esa_sampled["esa_class"] = esa_sampled["Map"].apply(lambda x: map_values(x, 40))
test_data["esa"] = pd.merge(test_data, esa_sampled, on=["lat", "lon"], how="left")["esa_class"]

glad = ee.ImageCollection("users/potapovpeter/Global_cropland_2019")
glad_results = glad.filterBounds(test_coll).map(lambda x: rasterExtraction(x, 30, test_coll)).flatten()
glad_sampled = geemap.ee_to_gdf(glad_results)
test_data["glad"] = pd.merge(test_data, glad_sampled, on=["lat", "lon"], how="left")["b1"]

# Then metrics only need to be ran once
results = {"Kenya": pd.DataFrame(), "Togo": pd.DataFrame()}
for country, df in test_data.groubpy("country"):
    for dataset in ["esa", "glad"]:
        report = classification_report(df["test_class"], df[dataset])
        results[country] = report_to_row(dataset, report, results[country])
</pre><p> </p><p>Let me know what you think!</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/nasaharvest/crop-mask/pull/228/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='45'/>